src/main/kotlin/com/sourcegraph/cody/chat/ui/LLMDropdown.kt

```diff
@@ -0,0 +1,94 @@
+package com.sourcegraph.cody.chat.ui
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBox
+import com.intellij.ui.MutableCollectionComboBoxModel
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.sourcegraph.cody.agent.CodyAgent
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.ChatModelsParams
+import com.sourcegraph.cody.agent.protocol.ChatModelsResponse
+import com.sourcegraph.cody.chat.SessionId
+import com.sourcegraph.cody.config.CodyAccount.Companion.isEnterpriseAccount
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.cody.ui.ChatModel
+import com.sourcegraph.cody.ui.LLMComboBoxItem
+import com.sourcegraph.cody.ui.LLMComboBoxRenderer
+import com.sourcegraph.common.BrowserOpener
+import com.sourcegraph.common.CodyBundle
+import java.util.concurrent.TimeUnit
+
+class LLMDropdown(val project: Project, private val modelFromState: ChatModel?) :
+    ComboBox<LLMComboBoxItem>(MutableCollectionComboBoxModel()) {
+
+  private var firstMessageSent: Boolean = false
+
+  init {
+    renderer = LLMComboBoxRenderer(this)
+    if (modelFromState != null) {
+      addItem(LLMComboBoxItem(modelFromState.icon, modelFromState.displayName))
+    }
+
+    isEnabled = false
+  }
+
+  fun fetchAndUpdateModels(sessionId: SessionId) {
+    if (modelFromState != null) {
+      return
+    }
+
+    CodyAgentService.withAgent(project) { agent ->
+      val chatModels = agent.server.chatModels(ChatModelsParams(sessionId))
+      val response = chatModels.completeOnTimeout(null, 4, TimeUnit.SECONDS).get()
+      val isCurrentUserPro = isCurrentUserPro(agent)
+      (renderer as LLMComboBoxRenderer).isCurrentUserFree = !isCurrentUserPro
```

Comment:
That is not strictly true, right?
If user is not pro, it can be either free or enterprise?

Reply:
fixed